editor/components/post-permalink/index.js

```diff
@@ -14,55 +14,151 @@ import { Dashicon, ClipboardButton, Button } from '@wordpress/components';
  * Internal Dependencies
  */
 import './style.scss';
-import { isEditedPostNew, getEditedPostAttribute } from '../../store/selectors';
+import { isEditedPostNew, getEditedPostAttribute, getEditedPostPreviewLink } from '../../store/selectors';
+import { editPost } from '../../store/actions';
+import { getWPAdminURL } from '../../utils/url';
 
 class PostPermalink extends Component {
 	constructor() {
 		super( ...arguments );
+
 		this.state = {
-			showCopyConfirmation: false,
+			iconClass: '',
+			samplePermalink: '',
+			samplePermalinkPrefix: '',
+			samplePermalinkSuffix: '',
+			postName: '',
+			editedPostName: '',
+			editingPermalink: false,
 		};
-		this.onCopy = this.onCopy.bind( this );
-		this.onFinishCopy = this.onFinishCopy.bind( this );
+
+		this.onSavePermalink = this.onSavePermalink.bind( this );
 	}
 
-	componentWillUnmount() {
-		clearTimeout( this.dismissCopyConfirmation );
+	componentWillMount() {
+		if ( this.props.samplePermalinkData ) {
+			this.updateSamplePermalink( this.props.samplePermalinkData[ 0 ], this.props.samplePermalinkData[ 1 ] );
+		}
 	}
 
-	onCopy() {
+	componentWillReceiveProps( nextProps ) {
+		if ( this.props.samplePermalinkData !== nextProps.samplePermalinkData ) {
+			this.updateSamplePermalink( nextProps.samplePermalinkData[ 0 ], nextProps.samplePermalinkData[ 1 ] );
+		}
+	}
+
+	/**
+	 * Updates the samplePermalink data in the state.
+	 *
+	 * @param {string} template Permalink structure to replace the tags in.
+	 * @param {string} postName The post slug to use in replacements.
+	 */
+	updateSamplePermalink( template, postName ) {
+		const regex = /%(?:postname|pagename)%/;
 		this.setState( {
-			showCopyConfirmation: true,
+			samplePermalink: template.replace( regex, postName ),
+			samplePermalinkPrefix: template.split( regex )[ 0 ],
+			samplePermalinkSuffix: template.split( regex )[ 1 ],
+			postName: postName,
```

Comment:
Minor: This can be shortened to just `postName,` when property name and variable value match.

See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#New_notations_in_ECMAScript_2015

Reply:
👍🏻